### PR TITLE
Extend add_vector to support upgrades on re-add

### PR DESCRIPTION
**Pull request by roystgnr**
This way we can have multiple code components asking for the same vector and they all get a superset of the capabilities they ask for.

### CODE REVIEW

**Review by lindsayad — COMMENTED** [review]
Nice!

**Review by lindsayad — APPROVED** [review]

File: src/systems/system.C [hunk 1]
```diff
@@ -733,13 +733,64 @@ NumericVector<Number> & System::add_vector (std::string_view vec_name,
   // Return the vector if it is already there.
   auto it = this->_vectors.find(vec_name);
   if (it != this->_vectors.end())
-    return *it->second;
+    {
+      // If the projection setting has *upgraded*, change it.
```
Comment (jwpeterson) [inline src/systems/system.C:737]: We discussed marking `System::add_vector` parallel_only, right?

**Review by jwpeterson — APPROVED** [review]
I can't remember what lead to us having separate storage for the vector types even though the types are stored on the vectors themselves, but that definitely seems good to be rid of that :+1:

File: src/systems/system.C (hunk 1 above)
Comment (roystgnr) [reply src/systems/system.C:737]: Yeah; sorry it slipped my mind.

Looking through, there's quite a few methods of similar type that ought to be marked parallel_only(); I'll make a separate PR for the lot.

**Pull request merged**